### PR TITLE
refactor: set gas limit associated to the ZRC20 for cctx revert outbound

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Implement a method to get the gas limit for the revert tx for a CCTX (query the gas limit of the associated ZRC20 for the asset sent)

Closes: https://github.com/zeta-chain/node/issues/1314

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [x] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go [hunk 1]
```diff
@@ -160,17 +161,27 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
...
-					cctx.OutboundTxParams = append(cctx.OutboundTxParams, &types.OutboundTxParams{
-						Receiver:        cctx.InboundTxParams.Sender,
-						ReceiverChainId: cctx.InboundTxParams.SenderChainId,
-						Amount:          cctx.InboundTxParams.Amount,
-						CoinType:        cctx.InboundTxParams.CoinType,
-						// NOTE(pwu): revert gas limit = initial outbound gas limit set by user
-						//TODO: determine a specific revert gas limit https://github.com/zeta-chain/node/issues/1065
-						OutboundTxGasLimit: cctx.OutboundTxParams[0].OutboundTxGasLimit,
-					})
-					err := k.PayGasAndUpdateCctx(
```
Comment (brewmaster012) [inline x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go:173]: does this remove the payment of gas?

Comment (lumtis) [reply x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go:173]: What do you mean? The function is still called.

Comment (brewmaster012) [reply x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go:173]: oh yeah my bad missed that.

**Review by brewmaster012 — APPROVED** [review]
LGTM

**Review by ws4charlie — APPROVED** [review]

**Pull request merged**